extensions/master_extension/v1/win-e2e-master-extension.sh

```diff
@@ -1,42 +1,16 @@
 #!/bin/bash
 
+echo "start master extension" >> /tmp/master_extension.log
+
 KUBECONFIG="$(find /home/*/.kube/config)"
 KUBECTL="kubectl --kubeconfig=${KUBECONFIG}"
 
-# Wait for coredns pod to be ready
-while true; do
-	COREDNS_READY=$(${KUBECTL} get pods -l k8s-app=kube-dns -n kube-system -o custom-columns=STATUS:status.containerStatuses[0].ready --no-headers)
-	if [ "${COREDNS_READY}" = "true" ]; then
-		echo "$(date -R) - coredns is ready" >> /tmp/master_extension.log
-		${KUBECTL} get pods --all-namespaces >> /tmp/master_extension.log
-		break
-	fi
-	sleep 2
-done
-
-# Wait for all pods in kube-system namespace to be ready before applying taints
-while true; do
-	NOTREADY_PODS=$(${KUBECTL} get pods -n kube-system -o custom-columns=STATUS:status.containerStatuses[0].ready --no-headers | grep -v "true")
-	if [ -z "${NOTREADY_PODS}" ]; then
-		echo "$(date -R) - All kube-system pods are ready" >> /tmp/master_extension.log
-		${KUBECTL} get pods --all-namespaces >> /tmp/master_extension.log
-		break
-	fi
-	sleep 2
-done
+${KUBECTL} wait --for=condition=ready pod --all -n kube-system --timeout 10m >> /tmp/master_extension.log
```

Comment:
there is an issue with using timeout with kubectl: https://github.com/kubernetes/kubectl/issues/754

This will end up waiting much longer than might be expected in worse case: # of system pods * 10mins

Reply:
This might not be an issue as before it would have been stuck in an infinite loop if pods didn't become ready.

@adelina-t  do you have any input?